lib/ldap.c

```diff
@@ -182,6 +182,41 @@ const struct Curl_handler Curl_handler_ldaps = {
 };
 #endif
 
+#ifdef USE_WIN32_LDAP
+static int Win_ldap_bind(struct connectdata *conn, LDAP *server, TCHAR *user, TCHAR *passwd)
+{
+    int rc = 0;
+    ULONG method = LDAP_AUTH_NEGOTIATE;
+    SEC_WINNT_AUTH_IDENTITY cred = { 0 };
+    PCHAR inuser = NULL;
+    PCHAR inpass = NULL;
+
+    if (user != NULL && passwd != NULL) {
+        cred.User = user;
+        cred.UserLength = strlen(user);
+        cred.Password = passwd;
+        cred.PasswordLength = strlen(passwd);
+        cred.Flags = SEC_WINNT_AUTH_IDENTITY_ANSI;
```

Comment:
the doc for ldap_bind_s is weird it says it uses PCHAR but I can see in Winldap.h:
~~~
WINLDAPAPI ULONG LDAPAPI ldap_bind_sW( LDAP *ld, __in_opt PWCHAR dn, __in_opt PWCHAR cred, ULONG method );
WINLDAPAPI ULONG LDAPAPI ldap_bind_sA( LDAP *ld, __in_opt PCHAR dn, __in_opt PCHAR cred, ULONG method );
~~~
and later
~~~
#if LDAP_UNICODE
[...]
#define ldap_bind_s ldap_bind_sW
[...]
#else
[...]
WINLDAPAPI ULONG LDAPAPI ldap_bind_s( LDAP *ld, __in_opt const PCHAR dn, __in_opt const PCHAR cred, ULONG method );
[...]
#endif
~~~
I don't know if LDAP_UNICODE is tied to UNICODE but I'd guess so.

> curl does not currently support Windows Unicode

That's a tricky one. Apparently it's possible to build curl with UNICODE, and that may enable some features otherwise not available. This is why some functions use TCHARs.

> And yes, should be _tcslen instead of strlen. Will fix it shortly.

Also change `if (` to `if(`, run checksrc please.

As to whether this is a good idea to add I don't know since I don't use LDAP protocol, so I think any +1 should come from someone else. How is the behavior different from Linux with LDAP after this change?